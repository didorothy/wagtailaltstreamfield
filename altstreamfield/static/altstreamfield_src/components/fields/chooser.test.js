import React from "react";
import { render, fireEvent, wait, waitForDomChange, waitForElementToBeRemoved, cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import Chooser from "./chooser";

describe('Chooser', () => {
    test("#minimal", () => {
        let handlers = {};
        let container = render(
            <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" onChosen={() => {}} />
        );
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#chosen_value', () => {
        let handlers = {};
        let container = render(
            <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" value={1} onChosen={() => {}}>
                <span>Chosen Value</span>
            </Chooser>
        );
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#chosen_required', () => {
        let handlers = {};
        let container = render(
            <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" value={1} required={true} onChosen={() => {}}>
                <span>Chosen Value</span>
            </Chooser>
        );
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#chosen_editUrl', () => {
        let handlers = {};
        let container = render(
            <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" value={1} required={true} editUrl="/url/1" onChosen={() => {}}>
                <span>Chosen Value</span>
            </Chooser>
        );
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('#chosen_className', () => {
        let handlers = {};
        let container = render(
            <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" value={1} required={true} editUrl="/url/1" className="test" onChosen={() => {}}>
                <span>Chosen Value</span>
            </Chooser>
        );
        expect(container.container).toMatchSnapshot();
        cleanup();
    });

    test('it can clear choice when not required', () => {
        let handlers = {};
        let last_value = 1
        let onChosen = (value) => {
            last_value = value;
        };
        let container = render(
            <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" value={1} onChosen={onChosen}>
                <span>Chosen Value</span>
            </Chooser>
        );
        expect(last_value).toEqual(1);
        let button = container.getByText('Clear choice');
        fireEvent.click(button);
        expect(last_value).toBeNull()
        cleanup();
    });

    test('it shows the modal workflow and handles a chosen event', () => {
        let handlers = {type: 'test'};
        let last_value = 1
        let onChosen = (value) => {
            last_value = value;
        };
        window.ModalWorkflow = (opts) => {
            opts.responses.testChosen({id: 3});
        }
        try {
            let container = render(
                <Chooser url="/chooser-url" modalOnLoadHandlers={handlers} chooserType="test" inputId="test" value={0} onChosen={onChosen}/>
            );
            expect(last_value).toEqual(1);
            let button = container.getByText('Choose a test');
            fireEvent.click(button);
            expect(last_value).toEqual(3);
        } finally {
            delete window.ModalWorkflow;
        }
        cleanup();
    });
});