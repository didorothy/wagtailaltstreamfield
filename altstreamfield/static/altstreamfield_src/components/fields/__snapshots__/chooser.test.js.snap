// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chooser #chosen_className 1`] = `
<div>
  <div
    class="chooser test"
  >
    <div
      class="chosen"
    >
      <span>
        Chosen Value
      </span>
      <ul
        class="actions"
      >
        <li>
          <button
            class="button action-choose button-small button-secondary"
            type="button"
          >
            Choose another 
            test
          </button>
        </li>
        <li>
          <a
            class="edit-link button button-small button-secondary"
            href="/url/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Edit this 
            test
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Chooser #chosen_editUrl 1`] = `
<div>
  <div
    class="chooser"
  >
    <div
      class="chosen"
    >
      <span>
        Chosen Value
      </span>
      <ul
        class="actions"
      >
        <li>
          <button
            class="button action-choose button-small button-secondary"
            type="button"
          >
            Choose another 
            test
          </button>
        </li>
        <li>
          <a
            class="edit-link button button-small button-secondary"
            href="/url/1"
            rel="noopener noreferrer"
            target="_blank"
          >
            Edit this 
            test
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Chooser #chosen_required 1`] = `
<div>
  <div
    class="chooser"
  >
    <div
      class="chosen"
    >
      <span>
        Chosen Value
      </span>
      <ul
        class="actions"
      >
        <li>
          <button
            class="button action-choose button-small button-secondary"
            type="button"
          >
            Choose another 
            test
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Chooser #chosen_value 1`] = `
<div>
  <div
    class="chooser"
  >
    <div
      class="chosen"
    >
      <span>
        Chosen Value
      </span>
      <ul
        class="actions"
      >
        <li>
          <button
            class="button action-clear button-small button-secondary"
            type="button"
          >
            Clear choice
          </button>
        </li>
        <li>
          <button
            class="button action-choose button-small button-secondary"
            type="button"
          >
            Choose another 
            test
          </button>
        </li>
      </ul>
    </div>
  </div>
</div>
`;

exports[`Chooser #minimal 1`] = `
<div>
  <div
    class="chooser blank"
  >
    <div
      class="unchosen"
    >
      <button
        class="button action-choose button-small button-secondary"
        id="test"
        type="button"
      >
        Choose 
        a
         
        test
      </button>
    </div>
  </div>
</div>
`;
